src/scripts/main.js

```diff
@@ -1,3 +1,191 @@
 'use strict';
 
-// write code here
+const headers = document.querySelector('thead');
+const tbody = document.querySelector('tbody');
+let colIndex = -1;
+
+headers.addEventListener('click', (e) => {
+  const target = e.target;
+  const index = target.cellIndex;
+
+  sortingTable(index, target.innerText, colIndex === index);
+
+  colIndex = (colIndex === index) ? -1 : index;
+});
+
+function sortingTable(index, nameColumn, directSorting) {
+  const sortTable = [...tbody.children];
+
+  sortTable.sort((first, second) => {
+    const a = first.children[index].innerText;
+    const b = second.children[index].innerText;
+
+    switch (nameColumn) {
+      case 'Name':
+      case 'Position':
+      case 'Office':
+        return a.localeCompare(b);
+
+      case 'Age':
+      case 'Salary':
+        return toNormalNumber(a) - toNormalNumber(b);
+    }
+  });
+
+  if (directSorting) {
+    sortTable.reverse();
+  }
+
+  tbody.append(...sortTable);
+}
+
+function toNormalNumber(string) {
+  let res = '';
+
+  string.includes('$')
+    ? res = string.slice(1).split(',').join('')
+    : res = string;
+
+  return Number(res);
+};
+
+[...tbody.children].forEach(tr => {
+  tr.addEventListener('click', () => {
+    [...tbody.children].forEach(row =>
+      row.classList.contains('active') ? row.classList.remove('active') : 1);
+
+    tr.classList.add('active');
+  });
+});
```

Comment:
separate it to different helper functions. Avoid such nesting in code. It's hard to read